experiments/capture-build-scan-info.gradle

```
@@ -0,0 +1,49 @@
+import org.gradle.util.GradleVersion
+
+initscript {
+    def pluginVersion = "3.6"
+
+    repositories {
+        gradlePluginPortal()
+    }
+
+    dependencies {
+        classpath("com.gradle:gradle-enterprise-gradle-plugin:${pluginVersion}")
+    }
+}
+
+def isTopLevelBuild = gradle.getParent() == null
+
+if (isTopLevelBuild) {
+    def version = GradleVersion.current().baseVersion
+    def atLeastGradle5 = version >= GradleVersion.version("5.0")
+    def atLeastGradle6 = version >= GradleVersion.version("6.0")
+    
+    if (atLeastGradle6) {
+        settingsEvaluated {
+            it.extensions["gradleEnterprise"].with {
```

i think you can just replace this like so?
```suggestion
            gradleEnterprise {
```

---

Weirdly, when I tried that, it didn't work. I get an error like:

```
Could not find method gradleEnterprise() for arguments [capture_build_scan_info_7nzf7873k91ckmu9lx8hsbm6z$_run_closure1$_closure3@25f76ec3] on build of type org.gradle.invocation.DefaultGradle.
```

I always have trouble getting my head around the way in which variables are scoped and such in an init script. There's so many nested closures that run at different times and are fed different contexts.....

---

Nit: I think a named closure delegate is more readable:
```
settingsEvaluated { settings ->
   settings.extensions["gradleEnterprise"]... 
```

---

I like this. It helps to clarify what object is actually available and in-scope.